include/boost/url/grammar/all_chars.hpp

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright (c) 2021 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_GRAMMAR_ALL_CHARS_HPP
+#define BOOST_URL_GRAMMAR_ALL_CHARS_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/grammar/detail/charset.hpp>
+
+namespace boost {
+namespace urls {
+namespace grammar {
+
+/** The set of all characters
+
+    @par Example
+    Character sets are used with rules and the
+    functions @ref find_if and @ref find_if_not.
+    @code
+    result< string_view > rv = parse( "JohnDoe", token_rule( all_chars ) );
+    @endcode
+
+    @par BNF
+    @code
+    ALL       = %x00-FF
+    @endcode
+
+    @see
+        @ref find_if,
+        @ref find_if_not,
+        @ref parse,
+        @ref token_rule.
+*/
+#ifdef BOOST_URL_DOCS
+constexpr __implementation_defined__ all_chars;
+#else
+struct all_chars_t
+{
+    constexpr
+    all_chars_t() noexcept = default;
+
+    constexpr
+    bool
+    operator()(char c) const noexcept
+    {
+        return c != '%';
```

Comment:
you can't do this in a type named `all_chars_t`

Reply:
Maybe we should do this instead:
```
template< class CharSet =
#ifdef BOOST_URL_DOCS
    __implementation_defined__
#else
    detail::decode_chars_t
#endif
>
auto decode(...
```

Reply:
Yes. I'll fix `decode` already.